deployarchive.ps1

```diff
@@ -1,16 +1,32 @@
 $build = (Get-Content .\build.json | Out-String | ConvertFrom-Json)
+
+function CreateZip {
+  param( [string] $name ) 
+
+  Add-Type -Assembly System.IO.Compression.FileSystem
+  $compressionLevel = [System.IO.Compression.CompressionLevel]::Optimal
+  $sourcedir = $env:BUILD_ARTIFACTSTAGINGDIRECTORY + "\" + $name
+  $zipfilename = $env:BUILD_ARTIFACTSTAGINGDIRECTORY + "\" + $name + ".zip"
+
+  [System.IO.Compression.ZipFile]::CreateFromDirectory($sourcedir, $zipfilename, $compressionLevel, $false)
+  if ($LASTEXITCODE -eq 1)
+  {
+    Write-Host "Error build project $_"
```

Comment:
Will this still work?  I don't think the $_ has any context here.  You'll probably just want this to be $name.